packages/react-dom/src/client/DOMPropertyOperations.js

```diff
@@ -172,3 +180,17 @@ export function setValueForProperty(
     }
   }
 }
+
+// Only used in DEV, stringifies a value and Warns if it took too long
+const stringifyWithPerformanceWarning = value => {
+  const stringifyStart = Date.now();
+  let attributeValue = (value: any).toString();
+  const stringifyEnd = Date.now();
+
+  warning(
+    stringifyEnd - stringifyStart <= 2,
+    'Stringifying your attribute is causing perfomance issues',
```

Comment:
Typo: pefomance => performance.

Reply:
The messaging here isn't clear enough. Let's also pass the attribute name to `stringifyWithPerformanceWarning`, improve the wording, and also include a component stack. You can get a component stack by importing and using `getCurrentFiberStackAddendum` from `ReactDebugCurrentFiber`.

Lets make it look like:

```js
warning(
  stringifyEnd - stringifyStart <= 2,
  'The attribute `%s` took more than 2 milliseconds to stringify. This usually means you provided a large object ' +
  'as the value for a DOM attribute, which can lead to performance issues.%s',
  attributeName,
  getCurrentFiberStackAddendum(),
)
```